decentraland/screens/BuyLandScreen.test.js

```diff
@@ -28,10 +29,11 @@ describe("BuyLandScreen", () => {
             myAssets={myAssets}
             selectedLandToBuy={estateForSale}
             removeLandForSale={removeLandForSale}
-            addToMyAssetsList={addToMyAssetsList}
-            removeMyAssetFromList={removeMyAssetFromList}
+            prependToMyAssetsList={prependToMyAssetsList}
+            removeFromMyAssetsList={removeFromMyAssetsList}
             prependLandForSaleToList={prependLandForSaleToList}
             setActionIdForMyAsset={setActionIdForMyAsset}
+            updateMyAssetStatus={updateMyAssetStatus}
```

Comment:
It's starting to feel odd the number of functions we're needing to pass through to this component

Reply:
Yes, what is the best way to handle with this? Merge some actions?
Example:
```javascript
/// from
    removeLandForSale(landForSale);
    prependToMyAssetsList({ ...asset, status: BUYING });
// to
    moveAssetFromForSaleToMyAssets(landForSale, { ...asset, status: BUYING });
```

I'm not sure how the best way having an action changing more than one reducer (and if it's desirable). I'll take a look.

Reply:
Hmm, let's see. I'm not sure why we would need:

`myAssets`

on the buy land screen.

Then I think this component can have one function called "process buy" for a piece of land.

Then that function up in whichever component / screen it's originally defined can handle splitting out into multiple actions for all the relevant state that's updated (removeLandForSale, prependToMyAssetsList, and setActionIdForMyAsset, and updateMyAssetStatus).

And for `removeFromMyAssetsList`, I'm not sure from first principles why that would be used when buying land anyway.

There's still only one reducer affected by any given action, but multiple actions are dispatched.

Reply:
Hmm, only some of what I just said makes sense. I forgot this was the top level screen. I guess there are some functions that maybe should be defined outside of the scope of a given screen though, so they can react even if the user navigates away in the meantime?

Reply:
So many of those functions are wrapped into the _buy function. I just didn't realize that the test would need them all defined in this way. Does it need to be passed in like this in the test for it to work?

Reply:
We can remove action functions from Proptype requirements, if we do so, it won't be needed.

Reply:
Hmm yeah, what value do we get out of `mapDispatchToProps`? It's possible we get some - I'm just not used to that approach.

Reply:
When we want to test some action behavior we still are able to pass it as props even if it wasn't on the `proptype` requirements. The only `pros` that I see is to have more strict component props definition. But I think worth more having less boilerplate.

Reply:
Yeah, if we were using them in the test it would make sense - and maybe that's a change we should make in the future and add an issue for. But for now the enzyme test isn't using them in any way https://github.com/tasitlabs/tasit/blob/766f5a4e8386e76ac4b7ec1e4a701e8bd67adfad/decentraland/screens/BuyLandScreen.test.js

Reply:
I think we can have a strict component props definition for the top-level buy function but not the functions that function uses. If we ever want to test the subfuntctions with mocks, we can compile a mock version of the buy function from mocked subfunctions. Does that make sense?

Reply:
I'm not sure since all functions from redux are passed as component prop separately.

Reply:
Oh wait, I've just realize that we are validating proptypes on `eslint` setup and if we'll just remove the action frunctions from we'll need to change the eslint config. I'll take a look if has a recommendation for the situation like ours.

Reply:
> I'm not sure since all functions from redux are passed as component prop separately.

Do they need to be?

Reply:
> Oh wait, I've just realize that we are validating proptypes on `eslint` setup and if we'll just remove the action functions from we'll need to change the eslint config. I'll take a look if has a recommendation for the situation like ours.

This is a naive question, but can we stop mapping dispatch to props?

Reply:
Otherwise, maybe this is "code smell" that this really is just one "action", and our reducers + datastore have been designed in too complicated of a way and need to find a way to handle this single action.

Reply:
Hmmmm maybe:
```javascript
const mapDispatchToProps = {
  removeLandForSale,
  prependLandForSaleToList,
  prependToMyAssetsList,
  removeFromMyAssetsList,
  setActionIdForMyAsset,
  updateMyAssetStatus,
};

// could be ?
const mapDispatchToProps = {
   buyFunctions: {
     removeLandForSale,
     prependLandForSaleToList,
     prependToMyAssetsList,
     removeFromMyAssetsList,
     setActionIdForMyAsset,
     updateMyAssetStatus, 
  }
};
```

Reply:
What do you think about multiple reducers reacting to a single action instead?

Reply:
There's an egghead lesson about it, so it must not be too much of an antipattern

https://egghead.io/lessons/react-respond-to-a-single-redux-action-in-multiple-reducers

Reply:
Hmmm, I'll take a look on that.

Reply:
I'm thinking we can make a new issue for this and merge this PR without making this change.

Reply:
Extracted to: https://github.com/tasitlabs/tasit/issues/341

Reply:
Thanks. Resolving this.